### PR TITLE
FOUNDATIONS: Retrieve Intern By Id

### PR DESCRIPTION
**Pull request by ADILATIC**
Closes #75

### CODE REVIEW

File: InternTrack.Core.Api/Services/Foundations/Interns/InternService.Exceptions.cs [hunk 1]
```diff
@@ -34,6 +34,10 @@ private async ValueTask<Intern> TryCatch(ReturningInternFuction returningInternF
             {
                 throw CreateAndLogValidationException(invalidInternException);
             }
+            catch (NotFoundInternException NotfoundInternException)
```
Comment (hassanhabib) [inline InternTrack.Core.Api/Services/Foundations/Interns/InternService.Exceptions.cs:37]: `NotFoundException` shouldn't come before `SqlException` - honor the chronological order here

File: InternTrack.Core.Api/Services/Foundations/Interns/InternService.Validations.cs [hunk 2]
```diff
@@ -107,6 +110,14 @@ private bool IsDateNotRecent(DateTimeOffset date)
             return timeDifference.Duration() > oneMinute;
         }
 
+        private void ValidateStorageIntern(Intern maybeIntern, Guid internId)
```
Comment (hassanhabib) [inline InternTrack.Core.Api/Services/Foundations/Interns/InternService.Validations.cs:113]: Methods that are used outside of the file take precedence over methods that are used within the file.

File: InternTrack.Core.Api/Services/Foundations/Interns/InternService.cs [hunk 3]
```diff
@@ -36,7 +37,20 @@ public partial class InternService : IInternService
             return await this.storageBroker.InsertInternAsync(intern);
         });
 
+        public ValueTask<Intern> RetrieveInternByIdAsync(Guid internId) =>
+           TryCatch(async () =>
```
Comment (hassanhabib) [inline InternTrack.Core.Api/Services/Foundations/Interns/InternService.cs:41]: `TryCatch` should be at the same level as `public`

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Logic.RetrieveById.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        public async Task ShouldRetrieveInternByIdAsync()
+        {
+            //given
+            Guid randomInternId = Guid.NewGuid();
+            Guid inputInternId = randomInternId;
+            Intern randomIntern = CreateRandomIntern();
+            Intern storageIntern = randomIntern;
+            Intern expectedIntern = storageIntern.DeepClone();
+
+            this.storageBrokerMock.Setup(broker => broker.SelectInternByIdAsync(randomInternId))
```
Comment (hassanhabib) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Logic.RetrieveById.cs:28]: This here doesn't match our style of indentation please check how we do it in other methods

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Logic.RetrieveById.cs [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+                .ReturnsAsync(storageIntern);
+
+            //when
```
Comment (hassanhabib) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Logic.RetrieveById.cs:31]: put one space here and for the rest `// when` do the same for ALL the rest

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Logic.RetrieveById.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            Intern actualIntern =
+                    await this.internService.RetrieveInternByIdAsync(inputInternId);
+
+            //then
+            actualIntern.Should().BeEquivalentTo(expectedIntern);
+
+            this.storageBrokerMock.Verify(broker =>
```
Comment (hassanhabib) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Logic.RetrieveById.cs:38]: This here also doesn't match the indentation style we follow across the application please review and update.
Do the same for the rest.

File: InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.RetrieveById.cs [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+            actualInternValidationException.Should().BeEquivalentTo(expectedInternValidationException);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionsAs(
+                    expectedInternValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectInternByIdAsync(It.IsAny<Guid>()),
+                  Times.Once);
```
Comment (hassanhabib) [inline InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.RetrieveById.cs:90]: This is not a tab - seems copied from somewhere. Make sure you *type* things out or else you gonna run into issues with spaces and tabs and such

File: InternTrack.Core.Api/InternTrack.Core.Api.csproj [hunk 8]
```diff
@@ -4,6 +4,13 @@
     <TargetFramework>net7.0</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (glhays) [inline InternTrack.Core.Api/InternTrack.Core.Api.csproj:7]: This needs to be removed

**Pull request merged**